pkg/api/types.go

```diff
@@ -859,8 +859,8 @@ func (o *OrchestratorProfile) GetAPIServerEtcdAPIVersion() string {
 	ret := "etcd3"
 	if o.KubernetesConfig != nil {
 		// if we are here, version has already been validated..
-		etcdversion, _ := semver.NewVersion(o.KubernetesConfig.EtcdVersion)
-		if etcdversion != nil && 2 == etcdversion.Major() {
```

Comment:
//TODO: re-add nil check

Reply:
done

Reply:
I remembered the reason I removed it is because `Make` will never return nil, it returns an empty Version if anything